src/parser.ts

```diff
@@ -0,0 +1,60 @@
+import { versionParserMap } from './parsers';
+
+function lookupVersion (array: string[], v: string): boolean {
+  return array.some(str => str.includes(`v${v}`) || str.includes(`${v}.`));
+}
+
+function retrieveBlockcertsVersion (context): number {
+  if (typeof context === 'string') {
+    context = [context];
+  }
+
+  const blockcertsContext = context.filter(ctx => typeof ctx === 'string').find(ctx => ctx.toLowerCase().indexOf('blockcerts') > 0);
+  const blockcertsContextArray: string[] = blockcertsContext.split('/').filter(str => str !== '');
+
+  const availableVersions: string[] = Object.keys(versionParserMap);
+
+  return parseInt(availableVersions.filter(version => lookupVersion(blockcertsContextArray, version.toString()))[0], 10);
+}
+
+interface ParsedCertificateValidityFormat {
+  isFormatValid: boolean;
+  error: string;
+}
+
+export interface ParsedCertificate extends ParsedCertificateValidityFormat {
```

Comment:
@lemoustachiste this is temporary as we should move that interface to the Certificate.ts file or other specific file. I didn't want to step over what's being worked on in the Explorer API branch.

Reply:
yeah agree